app/views/E2ESaveYourPasswordView.js

```diff
@@ -0,0 +1,168 @@
+	})
+
+	static propTypes = {
+		server: PropTypes.string,
+		navigation: PropTypes.object,
+		encryptionSetBanner: PropTypes.func,
+		theme: PropTypes.string
+	}
+
+	constructor(props) {
+		super(props);
+		this.mounted = false;
+		this.state = { password: '' };
+		this.init();
+	}
+
+	componentDidMount() {
+		this.mounted = true;
+	}
+
+	init = async() => {
+		const { server } = this.props;
+		try {
+			// Set stored password on local state
+			const password = await UserPreferences.getStringAsync(`${ server }-${ E2E_RANDOM_PASSWORD_KEY }`);
+			if (this.mounted) {
+				this.setState({ password });
+			} else {
+				this.state.password = password;
+			}
+		} catch {
+			// Do nothing
+		}
+	}
+
+	onSaved = async() => {
+		const { navigation, server, encryptionSetBanner } = this.props;
+		// Remove stored password
+		await UserPreferences.removeItem(`${ server }-${ E2E_RANDOM_PASSWORD_KEY }`);
+		// Hide encryption banner
+		encryptionSetBanner();
+		navigation.pop();
+	}
+
+	onCopy = () => {
+		const { password } = this.state;
+		Clipboard.setString(password);
+		EventEmitter.emit(LISTENER, { message: I18n.t('Copied_to_clipboard') });
+	}
+
+	onHowItWorks = () => {
+		const { navigation } = this.props;
+		navigation.navigate('E2EHowItWorksView');
+	}
+
+	render() {
+		const { password } = this.state;
+		const { theme } = this.props;
+
+		return (
+			<SafeAreaView theme={theme} style={{ backgroundColor: themes[theme].backgroundColor }}>
+				<StatusBar theme={theme} />
+				<ScrollView {...scrollPersistTaps} style={sharedStyles.container} contentContainerStyle={sharedStyles.containerScrollView}>
+					<View style={[styles.container, { backgroundColor: themes[theme].backgroundColor }]}>
+						<Text style={[styles.warning, { color: themes[theme].dangerColor }]}>{I18n.t('Save_Your_Encryption_Password_warning')}</Text>
+						<View style={styles.content}>
+							<Text style={[styles.passwordText, { color: themes[theme].bodyText }]}>{I18n.t('Your_password_is')}</Text>
+							<Text style={[styles.password, { color: themes[theme].bodyText }]}>{password}</Text>
+							<Button
+								onPress={this.onCopy}
+								style={[styles.copyButton, { backgroundColor: themes[theme].auxiliaryBackground }]}
+								title={I18n.t('Copy')}
+								type='secondary'
+								fontSize={12}
+								theme={theme}
+							/>
+						</View>
+						<Text style={[styles.info, { color: themes[theme].bodyText }]}>{I18n.t('Save_Your_Encryption_Password_info')}</Text>
+						<Button
+							onPress={this.onHowItWorks}
+							style={{ backgroundColor: themes[theme].auxiliaryBackground }}
+							title={I18n.t('How_It_Works')}
+							type='secondary'
+							theme={theme}
+						/>
+						<Button
+							onPress={this.onSaved}
+							title={I18n.t('I_Saved_My_E2E_Password')}
+							theme={theme}
+						/>
+					</View>
+				</ScrollView>
+			</SafeAreaView>
+		);
+	}
+}
+
+const mapStateToProps = state => ({
+	server: state.server.server
```

Comment:
server selector

Reply:
This screen will never appear on share extension, so, don't need to check the share extension current server, right?

Reply:
I've deleted the server selector for now, we can add this on other PR and put into all the screens, this we can continue as it is now, since this screen will never need to access the share extension current server.